src/Symfony/Component/Serializer/Tests/SerializerTest.php

```diff
@@ -1050,6 +1050,58 @@ public function testCollectDenormalizationErrorsWithConstructor()
 
         $this->assertSame($expected, $exceptionsAsArray);
     }
+
+    /** @requires PHP 8.0 */
+    public function testCollectDenormalizationErrorsWithConstructorAndTypeError()
+    {
+        $json = '{"bool": [true]}';
+
+        $classMetadataFactory = new ClassMetadataFactory(new AnnotationLoader(new AnnotationReader()));
+        $extractor = new PropertyInfoExtractor([], [new PhpDocExtractor(), new ReflectionExtractor()]);
+
+        $serializer = new Serializer(
+            [
+                new ObjectNormalizer($classMetadataFactory, null, null, $extractor, new ClassDiscriminatorFromClassMetadata($classMetadataFactory)),
+            ],
+            ['json' => new JsonEncoder()]
+        );
+
+        try {
+            $serializer->deserialize($json, Php80WithPromotedTypedConstructor::class, 'json', [
+                DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS => true,
+            ]);
+
+            $this->fail();
+        } catch (\Throwable $th) {
+            $this->assertInstanceOf(PartialDenormalizationException::class, $th);
+        }
+
+        $this->assertInstanceOf(Php80WithPromotedTypedConstructor::class, $th->getData());
+
+        $exceptionsAsArray = array_map(function (NotNormalizableValueException $e): array {
```

Comment:
Why do you convert the errors in an array? Couldn't you assert directly on the resulting error list to simplify the test?